pkg/gateway/model_build_targets_test.go

```diff
@@ -72,6 +81,38 @@ func Test_Targets(t *testing.T) {
 				},
 			},
 		},
+		{
```

Comment:
Can you add a unit test to test service delete ?

Reply:
Unit test is covered at line 85-115

Reply:
also, can u add a new test for just service add without HTTProute stuff...

Reply:
Created issue for adding e2etest: https://github.com/aws/aws-application-networking-k8s/issues/257